### PR TITLE
Add compatibility with Swift 5's Result

### PR DESCRIPTION
**Pull request by ikesyo**
This aims at providing a migration path for the Result type in the Swift 5's standard library.

- https://github.com/apple/swift-evolution/blob/master/proposals/0235-add-result.md
- https://forums.swift.org/t/accepted-with-modifications-se-0235-add-result-to-the-standard-library/18603

### CODE REVIEW

**Comment by norio-nomura** [comment]
Since `Package.swift` still uses `swift-tools-version:4.0`, SwiftPM uses `-swift-version 4` on compiling. It would be better to add `Package@swift-5.swift` using `swift-tools-version:5.0` for checking compatibility with `-swift-version 5`.

**Comment by ikesyo** [comment]
@norio-nomura Thanks for pointing it out! I'll address it.

File: Result/Result.swift [hunk 1]
```diff
@@ -140,7 +173,7 @@ public func materialize<T>(_ f: () throws -> T) -> Result<T, AnyError> {
 
 @available(*, deprecated, renamed: "Result.init(_:)")
 public func materialize<T>(_ f: @autoclosure () throws -> T) -> Result<T, AnyError> {
-	return Result(f)
+	return Result(try f())
```
Comment (ikesyo) [inline Result/Result.swift]: This is needed (for Swift 5) because of the error: `error: add () to forward @autoclosure parameter`.

**Comment by ikesyo** [comment]
Thanks!

**Comment by mdiep** [comment]
Awesome. 🤘🏻

I'm assuming we'll do a 5.0 release that removes everything that's defined in the Standard Library version.

**Comment by marchy** [comment]
@mdiep Do you anticipate this as a future release beyond the current type-aliased 4.1.0 release? Great to see the migration path here and the enablement which this library has produced  =)

**Comment by mdiep** [comment]
I suppose we should do this, yes. But I'm unclear if this should be a minor release or a major release.

**Comment by marchy** [comment]
Makes perfect sense. I suppose should go as a major release, even if the consuming APIs may not be breaking (which I'm not sure is the case or not).

Conceptually it's definitely a big thing, so a major release will at least capture/communicate that.

**Pull request merged**